pstoremem/addr_book.go

```diff
@@ -171,6 +240,21 @@ func (mab *memoryAddrBook) AddAddrs(p peer.ID, addrs []ma.Multiaddr, ttl time.Du
 			}
 		}
 	}
+
+	// when adding signed addrs, make sure only the addrs from the input list remain.
+	if signed {
```

Comment:
You should do the removal before. This can remove valid signed addresses if we're transitioning from unsigned to signed.